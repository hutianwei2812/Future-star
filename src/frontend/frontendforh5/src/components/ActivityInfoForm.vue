<template>
  <div>
    <title-bar :is-on-load="isOnLoad"></title-bar>
    <nav-bar :special-color="specialColor"></nav-bar>
    <div>
      <el-page-header @back="goBack" content></el-page-header>
      <div>
        <h1>活动详情</h1>
        <el-card :class="allow_change==0?'box-card':'box-card-allow'" shadow="hover">
          <div slot="header" class="clearfix">
          </div>
          <div class="bigForm">
            活动时间：&nbsp;&nbsp;&nbsp;&nbsp;
            <el-date-picker
              type="datetimerange"
              v-model="activity_time"
              range-separator="至"
              start-placeholder="开始日期"
              end-placeholder="结束日期"
              align="right"
            ></el-date-picker>
            &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
            当前活动状态：&nbsp;&nbsp;{{activity_status}}&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;活动内部编号：&nbsp;&nbsp;{{activityIndex}}
            <br />
            <br />活动目标人数：&nbsp;&nbsp;&nbsp;&nbsp;
            <el-input-number size="medium" v-model="activity_peopleneed" />
            &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
            已报名人数：&nbsp;&nbsp;{{activity_peopleSignUp}}&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
            <br />
            <br />活动名称：&nbsp;&nbsp;&nbsp;&nbsp;
            <el-input class="normalInputbox" v-model="activity_name" placeholder="请输入" />
            <br />
            <br />活动地点：&nbsp;&nbsp;&nbsp;&nbsp;
            <el-input class="normalInputbox" v-model="activity_location" placeholder="请输入"></el-input>
            <br />
            <br />活动描述：&nbsp;&nbsp;&nbsp;&nbsp;
            <el-input class="normalInputbox" v-model="activity_abstract" placeholder="请输入"></el-input>
            <br />
            <br /><span style="text-align: left">活动具体流程：</span>&nbsp;&nbsp;&nbsp;&nbsp;
            <div class="nextLine"></div>
            <el-input
              class="textField"
              type="textarea"
              :autosize="{ minRows: 5}"
              placeholder="请输入内容"
              v-model="activity_items"
            ></el-input>
            <br />
            <br />活动注意事项：&nbsp;&nbsp;&nbsp;&nbsp;
            <div class="nextLine"></div>
            <el-input
              class="textField"
              type="textarea"
              :autosize="{ minRows: 5}"
              placeholder="请输入内容"
              v-model="activity_tips"
            ></el-input>
            <br />
            <br />
            <el-collapse class="peoplesignup">
              <div class="AttendPeople">
                <el-collapse-item title="当前已报名人员">
                  <el-table :data="schoolMateInfo" stripe class="fillTable">
                    <el-table-column prop="name" label="姓名" width="200"></el-table-column>
                    <el-table-column prop="gender" label="性别" width="200"></el-table-column>
                    <el-table-column prop="phoneNumber" label="联系方式" width="349"></el-table-column>
                  </el-table>
                </el-collapse-item>
              </div>
            </el-collapse>
          </div>
        </el-card>
        <br />
        <br />
        <br />
        <br />
        <el-button
          type="primary"
          icon="el-icon-edit"
          class="operbutton0"
          @click="changePage()"
          v-if="allow_change==0"
        >修改内容</el-button>
        <el-button
          type="primary"
          icon="el-icon-check"
          class="operbutton"
          @click="modifyAct"
          v-if="allow_change==1"
        >保存修改</el-button>
        <el-button
          type="primary"
          icon="el-icon-close"
          class="operbutton2"
          @click="changePage()"
          v-if="allow_change==1"
        >取消修改</el-button>
        <br />
        <br />
        <br />
        <br />
      </div>
    </div>
  </div>
</template>

<script>
import { access } from 'fs'
import TitleBar from '../components/TitleBar.vue'
import NavBar from '../components/NavBar.vue'
import axios from 'axios'
import Qs from 'qs'
export default {
  name: 'ActivityInfoForm',
  props: {},
  data() {
    return {
      targetUrl: 'http://117.50.34.200:8002/specificActivityMsg/',
      activityIndex: 0,
      isOnLoad: true,
      specialColor: 3,
      allow_change: 0,
      activity_name: '',
      activity_time: [],
      activity_location: '',
      activity_abstract: '',
      activity_peopleneed: 0,
      activity_tips: '',
      activity_items: '',
      activity_status: '未开始',
      activity_peopleSignUp: 0,
      schoolMateInfo: []
    }
  },
  methods: {
    goBack() {
      this.$router.push({
        name: 'ActivityManagePage'
      })
    },
    modifyAct() {
      this.modifyActivity()
      this.changePage()
    },
    changePage() {
      let power = localStorage['powerActivity']
      if (power === 'true') {
        if (this.allow_change == 0) {
          this.allow_change = 1
        } else {
          this.allow_change = 0
        }
      } else if (power === 'false') {
        this.$message({
          type: 'warning',
          message: '权限不足!'
        })
      }
    },
    readSpecificActivity() {
      const _this = this
      axios({
        url: _this.targetUrl,
        method: 'post',
        data: {
          id: _this.activityIndex
        },
        transformRequest: [
          function(data) {
            return Qs.stringify(data)
          }
        ]
      }).then(response => {
        var data = response.data
        var startEnd = data[0].activity_time.split(',')
        this.activity_time.push(startEnd[0])
        this.activity_time.push(startEnd[1])
        var starttime = new Date(Date.parse(startEnd[0]))
        var endtime = new Date(Date.parse(startEnd[1]))
        if (new Date() < starttime) {
          this.activity_status = '未开始'
        } else if (new Date() < endtime) {
          this.activity_status = '进行中'
        } else {
          this.activity_status = '已结束'
        }
        this.activity_name = data[0].activity_name
        this.activity_location = data[0].activity_location
        this.activity_abstract = data[0].activity_abstract
        this.activity_peopleneed = data[0].activity_peopleneed
        this.activity_tips = data[0].activity_tips
        this.activity_items = data[0].activity_items
        this.activity_peopleSignUp = data[1]
      })
    },
    getSchoolInfo() {
      axios({
        url: 'http://117.50.34.200:8002/queryActivityAttenders/',
        method: 'post',
        data: {
          id: this.activityIndex
        },
        transformRequest: [
          function(data) {
            return Qs.stringify(data)
          }
        ]
      }).then(response => {
        let userData = response.data
        for (let index = 0; index < userData.length; index++) {
          let userGender = ''
          if (userData[index].user_gender === 'm') {
            userGender = '男'
          } else if (userData[index].user_gender === 'f') {
            userGender = '女'
          }
          this.$set(this.schoolMateInfo, index, {})
          this.$set(
            this.schoolMateInfo[index],
            'name',
            userData[index].user_name
          )
          this.$set(this.schoolMateInfo[index], 'gender', userGender)
          this.$set(
            this.schoolMateInfo[index],
            'phoneNumber',
            userData[index].user_telephone
          )
        }
      })
    },
    modifyActivity() {
      if (
        this.activity_name !== '' &&
        this.activity_time !== [] &&
        this.activity_location !== '' &&
        this.activity_abstract !== '' &&
        this.activity_peopleneed !== 0
      ) {
        axios({
          url: 'http://117.50.34.200:8002/modifyActivity/',
          method: 'post',
          data: {
            id: this.activityIndex,
            activity_name: this.activity_name,
            activity_time: this.activity_time[0] + ',' + this.activity_time[1],
            activity_location: this.activity_location,
            activity_abstract: this.activity_abstract,
            activity_peopleneed: this.activity_peopleneed,
            activity_tips: this.activity_tips,
            activity_status: '',
            activity_items: this.activity_items
          },
          transformRequest: [
            function(data) {
              return Qs.stringify(data)
            }
          ]
        }).then(response => {
          if (response.data.code === '1') {
            this.$message('活动信息修改成功！')
            this.activity_time = []
            this.readSpecificActivity()
          }
        })
      } else {
        this.$message('表单信息填写不全，请填写完毕后再提交！')
      }
    }
  },
  created() {
    let adminName = localStorage['name']
    let adminPwd = localStorage['pwd']
    if (typeof adminName !== 'undefined' && adminName !== '') {
      axios({
        url: 'http://117.50.34.200:8002/adminForLogin/',
        method: 'post',
        data: {
          admin_name: adminName,
          admin_password: adminPwd
        },
        transformRequest: [
          function(data) {
            return Qs.stringify(data)
          }
        ]
      }).then(response => {
        if (response.data.code === 'ok') {
          this.activityIndex = parseInt(sessionStorage['activityIndex'])
          this.readSpecificActivity()
          this.getSchoolInfo()
        } else if (response.data.code === 'no') {
          this.$store.commit('setName', '')
          this.$store.commit('setPwd', '')
          this.$store.commit('setPowerCamp', 'false')
          this.$store.commit('setPowerCourse', 'false')
          this.$store.commit('setPowerActivity', 'false')
          this.$store.commit('setPowerAdministrator', 'false')
          this.$router.push({
            name: 'Index'
          })
        } else {
          this.$notify,
            error({
              title: 'Error',
              message: response.data
            })
          this.$store.commit('setName', '')
          this.$store.commit('setPwd', '')
          this.$store.commit('setPowerCamp', 'false')
          this.$store.commit('setPowerCourse', 'false')
          this.$store.commit('setPowerActivity', 'false')
          this.$store.commit('setPowerAdministrator', 'false')
          this.$router.push({
            name: 'Index'
          })
        }
      })
    } else {
      this.$router.push({
        name: 'Index'
      })
    }
  },
  components: {
    TitleBar,
    NavBar
  }
}
</script>


<style scoped>
h1 {
  text-align: center;
}
.peoplesignup {
  pointer-events: auto !important;
}
.textField {
  width: 85%;
}
.normalInputbox {
  width: 75%;
}
.text {
  font-size: 20px;
}
.item {
  margin-bottom: 18px;
  height: 30px;
}

.clearfix:before,
.clearfix:after {
  display: table;
  content: '';
}
.clearfix:after {
  clear: both;
}
.box-card {
  top: 20px;
  width: 1100px;
  position: relative;
  margin: auto;
  pointer-events: none;
  background-color: rgb(255, 255, 255);
}
.box-card-allow {
  top: 20px;
  width: 1100px;
  position: relative;
  margin: auto;
  pointer-events: auto;
}
.operbutton0 {
  width: 160px;
  left: 45%;
  position: relative;
  margin: auto;
}
.operbutton {
  width: 160px;
  left: 40%;
  position: relative;
  margin: auto;
}
.operbutton2 {
  width: 160px;
  left: 41%;
  position: relative;
  margin: auto;
}
.AttendPeople {
  position: relative;
  margin: auto;
  width: 750px;
}
.bigForm {
  text-align: center;
}
.nextLine {
  margin: 10px 0;
}
.fillTable {
  width: 100%;
}
</style>