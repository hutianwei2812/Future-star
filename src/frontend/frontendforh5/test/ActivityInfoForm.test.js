import { mount } from '@vue/test-utils'
import ActivityInfoForm from '@/components/ActivityInfoForm'
import Vue from 'vue'
import ElementUI from 'element-ui'
Vue.use(ElementUI)

test('renders correctly', () => {
  const wrapper = mount(ActivityInfoForm)
  expect(wrapper.element).toMatchSnapshot()
})