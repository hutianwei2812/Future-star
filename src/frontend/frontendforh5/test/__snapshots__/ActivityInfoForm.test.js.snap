// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <header
    class="el-header"
    style="height: 60px;"
  >
    <span
      class="title-span"
    >
      未来之星管理系统
    </span>
     
    <span
      class="user-span"
    >
      
    </span>
     
    <a
      class="el-link el-link--primary is-underline"
    >
      <!---->
      <span
        class="el-link--inner"
      >
        注销
      </span>
      <!---->
    </a>
  </header>
   
  <div
    class="nav"
  >
    <section
      class="el-container is-vertical"
    >
      <header
        class="el-header"
        style="height: 60px;"
      >
        <div
          class="newDiv"
        >
          <ul
            class="sub_nav"
          >
            <div>
              <li
                class="sub_nav_li"
                id="normalColor"
              >
                报名管理
              </li>
            </div>
             
            <div>
              <li
                class="sub_nav_li"
                id="normalColor"
              >
                课程管理
              </li>
            </div>
             
            <div>
              <li
                class="sub_nav_li"
                id="blueColor"
              >
                活动管理
              </li>
            </div>
             
            <div>
              <li
                class="sub_nav_li"
                id="normalColor"
              >
                校友管理
              </li>
            </div>
             
            <div>
              <li
                class="sub_nav_li"
                id="normalColor"
              >
                管理员权限设置
              </li>
            </div>
          </ul>
        </div>
      </header>
    </section>
  </div>
   
  <div>
    <div
      class="el-page-header"
    >
      <div
        class="el-page-header__left"
      >
        <i
          class="el-icon-back"
        />
        <div
          class="el-page-header__title"
        >
          返回
        </div>
      </div>
      <div
        class="el-page-header__content"
      >
        活动详情
      </div>
    </div>
     
    <div>
      <div
        class="el-card is-hover-shadow box-card"
      >
        <div
          class="el-card__header"
        >
          <div
            class="clearfix"
          >
            <span>
              活动详情
            </span>
          </div>
        </div>
        <div
          class="el-card__body"
        >
           
          <div
            class="bigForm"
          >
            
          活动时间：    
          
            <div
              class="el-date-editor el-range-editor el-input__inner el-date-editor--datetimerange"
            >
              <i
                class="el-input__icon el-range__icon el-icon-time"
              />
              <input
                autocomplete="off"
                class="el-range-input"
                name=""
                placeholder="开始日期"
              />
              <span
                class="el-range-separator"
              >
                至
              </span>
              <input
                autocomplete="off"
                class="el-range-input"
                name=""
                placeholder="结束日期"
              />
              <i
                class="el-input__icon el-range__close-icon"
              />
            </div>
            
                 
          当前活动状态：  未开始          活动内部编号：  NaN
          
            <br />
             
            <br />
            活动目标人数：    
          
            <div
              class="el-input-number el-input-number--medium"
            >
              <span
                class="el-input-number__decrease"
                role="button"
              >
                <i
                  class="el-icon-minus"
                />
              </span>
              <span
                class="el-input-number__increase"
                role="button"
              >
                <i
                  class="el-icon-plus"
                />
              </span>
              <div
                class="el-input el-input--medium"
              >
                <!---->
                <input
                  aria-disabled="undefined"
                  aria-valuemax="Infinity"
                  aria-valuemin="-Infinity"
                  aria-valuenow="0"
                  autocomplete="off"
                  class="el-input__inner"
                  max="Infinity"
                  min="-Infinity"
                  role="spinbutton"
                  type="text"
                />
                <!---->
                <!---->
                <!---->
                <!---->
              </div>
            </div>
            
                          
          已报名人数：  0          
          
            <br />
             
            <br />
            活动名称：    
          
            <div
              class="normalInputbox el-input"
            >
              <!---->
              <input
                autocomplete="off"
                class="el-input__inner"
                placeholder="请输入"
                type="text"
              />
              <!---->
              <!---->
              <!---->
              <!---->
            </div>
             
            <br />
             
            <br />
            活动地点：    
          
            <div
              class="normalInputbox el-input"
            >
              <!---->
              <input
                autocomplete="off"
                class="el-input__inner"
                placeholder="请输入"
                type="text"
              />
              <!---->
              <!---->
              <!---->
              <!---->
            </div>
             
            <br />
             
            <br />
            活动描述：    
          
            <div
              class="normalInputbox el-input"
            >
              <!---->
              <input
                autocomplete="off"
                class="el-input__inner"
                placeholder="请输入"
                type="text"
              />
              <!---->
              <!---->
              <!---->
              <!---->
            </div>
             
            <br />
             
            <br />
            活动具体流程：    
          
            <div
              class="nextLine"
            />
             
            <div
              class="textField el-textarea"
            >
              <textarea
                autocomplete="off"
                class="el-textarea__inner"
                placeholder="请输入内容"
              />
              <!---->
            </div>
             
            <br />
             
            <br />
            活动注意事项：    
          
            <div
              class="nextLine"
            />
             
            <div
              class="textField el-textarea"
            >
              <textarea
                autocomplete="off"
                class="el-textarea__inner"
                placeholder="请输入内容"
              />
              <!---->
            </div>
             
            <br />
             
            <br />
             
            <div
              aria-multiselectable="true"
              class="el-collapse"
              role="tablist"
            >
              <div
                class="AttendPeople"
              >
                <div
                  class="el-collapse-item"
                >
                  <div
                    aria-controls="el-collapse-content-1675"
                    aria-describedby="el-collapse-content-1675"
                    role="tab"
                  >
                    <div
                      class="el-collapse-item__header"
                      id="el-collapse-head-1675"
                      role="button"
                      tabindex="0"
                    >
                      当前已报名人员
                      <i
                        class="el-collapse-item__arrow el-icon-arrow-right"
                      />
                    </div>
                  </div>
                  <div
                    aria-hidden="true"
                    aria-labelledby="el-collapse-head-1675"
                    class="el-collapse-item__wrap"
                    id="el-collapse-content-1675"
                    role="tabpanel"
                    style="display: none;"
                  >
                    <div
                      class="el-collapse-item__content"
                    >
                      <div
                        class="el-table fillTable el-table--fit el-table--striped el-table--enable-row-hover el-table--enable-row-transition"
                      >
                        <div
                          class="hidden-columns"
                        >
                          <div />
                           
                          <div />
                           
                          <div />
                        </div>
                        <div
                          class="el-table__header-wrapper"
                        >
                          <table
                            border="0"
                            cellpadding="0"
                            cellspacing="0"
                            class="el-table__header"
                          >
                            <colgroup />
                            <thead
                              class=""
                            >
                              <tr
                                class=""
                              />
                            </thead>
                          </table>
                        </div>
                        <div
                          class="el-table__body-wrapper is-scrolling-none"
                        >
                          <table
                            border="0"
                            cellpadding="0"
                            cellspacing="0"
                            class="el-table__body"
                          >
                            <colgroup />
                            <tbody>
                              <tr
                                class="el-table__row"
                              />
                              <tr
                                class="el-table__row el-table__row--striped"
                              />
                              <tr
                                class="el-table__row"
                              />
                              <tr
                                class="el-table__row el-table__row--striped"
                              />
                              <!---->
                            </tbody>
                          </table>
                          <!---->
                          <!---->
                        </div>
                        <!---->
                        <!---->
                        <!---->
                        <!---->
                        <div
                          class="el-table__column-resize-proxy"
                          style="display: none;"
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
       
      <br />
       
      <br />
       
      <br />
       
      <br />
       
      <button
        class="el-button operbutton0 el-button--primary"
        type="button"
      >
        <!---->
        <i
          class="el-icon-edit"
        />
        <span>
          修改内容
        </span>
      </button>
       
      <!---->
       
      <!---->
       
      <br />
       
      <br />
       
      <br />
       
      <br />
    </div>
  </div>
</div>
`;
